### ISSUE TITLE
Option for pretty (extension-less) routes

### ISSUE THREAD

**Issue by nobkd** [nuejs/nue#282]
We have a file `<project-root>/x.md` which will currently get transformed to `/x.html` on the web.

Allow transformation to `/x` or `/x/` by putting the resulting file to `/x/index.html`

**Comment by korabdoroci** [comment]
Wouldn't it be simpler to just have a tag in the frontmatter instead?

**Comment by nobkd** [comment]
Hmm. But what if I want that behavior for all pages, not just select ones?
Then I would have to add the resulting file path in the front matter of each file.
I can understand the pros of that approach, but that would be great to have additionally to the option for pretty links (where I don't have to care about the file extension of pages)

(it'll probably be hard enough to keep the paths to the assets of a page correct)

**Comment by korabdoroci** [comment]
I was thinking something like: "pretty_routes: true" in the site.yaml which would then apply to the whole site. You could also do it in each app.yaml.

**Comment by nobkd** [comment]
I was actually thinking the same and wanted a config option for `site.yaml`.
But I probably misunderstood, because of mentioning front-matter (which is `yaml` config at the start of a `md` page to my knowledge)

**Comment by korabdoroci** [comment]
Yes, I was unclear. I think this would be a great addition to Nue.
